src/gui/advancedsettings.cpp

```diff
@@ -476,6 +483,14 @@ void AdvancedSettings::loadAdvancedSettings()
     m_spinBoxOutgoingPortsMax.setMaximum(65535);
     m_spinBoxOutgoingPortsMax.setValue(session->outgoingPortsMax());
     addRow(OUTGOING_PORT_MAX, tr("Outgoing ports (Max) [0: Disabled]"), &m_spinBoxOutgoingPortsMax);
+#if (LIBTORRENT_VERSION_NUM >= 10206)
+    // UPnP lease duration
+    m_spinBoxUPnPLeaseDuration.setMinimum(0);
+    m_spinBoxUPnPLeaseDuration.setMaximum(std::numeric_limits<int>::max());
+    m_spinBoxUPnPLeaseDuration.setValue(session->upnp_lease_duration());
+    m_spinBoxUPnPLeaseDuration.setSuffix(tr(" s", " seconds"));
+    addRow(UPNP_LEASE_DURATION, tr("UPnP Lease Duration [0: Permanent Lease]"), &m_spinBoxUPnPLeaseDuration);
```

Comment:
I think we can make use of [`setSpecialValueText()`](https://doc.qt.io/qt-5/qabstractspinbox.html#specialValueText-prop).

BTW, I think all of the advanced options that has special value at minimum should be using it, but that is for another PR...

Reply:
I'm not opposed to that. I think in doing so, each feature should be linked to documentation (not possible yet since it's unmerged) or have some form of tooltip/reference that states that the option is available, as Joe qBit may not know the option is there just by looking at the spinbox.

Reply:
>I'm not opposed to that. I think in doing so, each feature should be linked to documentation (not possible yet since it's unmerged) or have some form of tooltip/reference that states that the option is available, as Joe qBit may not know the option is there just by looking at the spinbox.

Agreed, please discard my suggestion. Lets keep the current form.

>each feature should be linked to documentation (not possible yet since it's unmerged)

I would guess it is going to be `https://www.libtorrent.org/reference-Settings.html#upnp_lease_duration` if without surprises.